### PR TITLE
feat:(TNLT-6350): sport lead2no pic2

### PR DESCRIPTION
**Pull request by justadamhopkins**
Draft of the lead two no pic and two variant

### CODE REVIEW

File: packages/section/src/utils/transformSlice.ts [hunk 1]
```diff
@@ -36,5 +46,57 @@ export const transformSlice = (isTablet: boolean, sectionTitle: string) => (
...
+  // no transform but base config is truthy so passes the base config to slice that contains a configured tile or tiles
+  if (!transformation && baseConfigs[slice.name]) {
+    return {
+      ...slice,
+      ...mergeBaseConfig,
+    };
+  }
+
+  // uses base config if someone forgets to set ovverrides in transform
+  if (!transformation.overrides)
```
Comment (c-j-j) [inline packages/section/src/utils/transformSlice.ts:75]: Can we have typescript prevent this situation? I.e. if you set a `transformation` then you have to provide an `overrides` field?

File: packages/edition-slices/src/slices/leadtwonopicandtwovariant2/baseLeadTwoNoPicAndTwoVariant2Config.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+export const baseLeadTwoNoPicAndTwoVariant2Config = {
```
[the remark this answers is not in the document]
Comment (aronshamash) [reply packages/edition-slices/src/slices/leadtwonopicandtwovariant2/baseLeadTwoNoPicAndTwoVariant2Config.ts:1]: I called this `config.ts` in my PR

File: packages/edition-slices/src/tiles/tile-a/index.js [hunk 3]
```diff
@@ -9,8 +9,12 @@ import {
 } from "../shared";
 import styles from "./styles";
 
-const TileA = ({ onPress, tile }) => {
-  const crop = getTileImage(tile, "crop169");
+const TileA = ({
+  onPress,
+  tile,
+  imageAspectRatio: { width = 16, height = 9 },
```
[the remark this answers is not in the document]
Comment (aronshamash) [reply packages/edition-slices/src/tiles/tile-a/index.js:15]: @justadamhopkins is this tile stay as old style tile ?

File: packages/edition-slices/src/slices/leadtwonopicandtwovariant2/baseLeadTwoNoPicAndTwoVariant2Config.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (justadamhopkins) [reply packages/edition-slices/src/slices/leadtwonopicandtwovariant2/baseLeadTwoNoPicAndTwoVariant2Config.ts:1]: sure I think baseConfig makes sense

File: packages/edition-slices/src/tiles/tile-a/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (justadamhopkins) [reply packages/edition-slices/src/tiles/tile-a/index.js:15]: yes i need to refactor this back to whatever is on master.

File: packages/section/src/section.tsx [hunk 4]
```diff
@@ -29,7 +30,12 @@ interface Props {
   onViewed: (item: any, slices: any[]) => void;
   publishedTime: string;
   receiveChildList: (childList: any) => void;
-  section: any;
+  section: {
+    title: sectionConfigs.SectionTitles;
+    cover: any;
+    name: any;
```
[the remark this answers is not in the document]
Comment (justadamhopkins) [reply packages/section/src/section.tsx:36]: done

File: packages/edition-slices/src/configured-tiles/tile-col-with-image-bottom/index.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+  getTileSummary,
+  TileImage,
+  TileLink,
+  TileSummary,
+  withTileTracking,
+} from "../../tiles/shared";
+import stylesFactory from "./styles";
+import WithoutWhiteSpace from "../../tiles/shared/without-white-space";
+import merge from "lodash.merge";
+import getAspectRatio from "../../utils/getAspectRatio";
```
Comment (aronshamash) [inline packages/edition-slices/src/configured-tiles/tile-col-with-image-bottom/index.tsx:14]: I have moved to `images/utils` in my PR

File: packages/section/src/utils/sectionConfigs.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  world: "World",
+  business: "Business",
+  sport: "Sport",
+  times2: "Times2",
+  register: "Register",
+  puzzles: "Puzzles",
+  law: "Law",
+  scotland: "Scotland",
+  ireland: "Ireland",
+} as const;
```
[the remark this answers is not in the document]
Comment (justadamhopkins) [reply packages/section/src/utils/sectionConfigs.ts:14]: done

File: packages/section/src/utils/transformSlice.ts (hunk 1 above)
Comment (justadamhopkins) [reply packages/section/src/utils/transformSlice.ts:75]: yes, i haven't bothered with types yet? However, this change still wouldn't stop the app being merged built etc so at this point not sure what is the safest approach?

File: packages/edition-slices/src/configured-tiles/tile-col-with-image-bottom/index.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    return (
+      <TileImage
+        aspectRatio={getAspectRatio(imageProps?.ratio)}
+        relativeWidth={relativeWidth}
+        relativeHeight={relativeHeight}
+        relativeHorizontalOffset={relativeHorizontalOffset}
+        relativeVerticalOffset={relativeVerticalOffset}
+        style={styles.imageContainer}
+        uri={url}
+        fill
```
[the remark this answers is not in the document]
Comment (justadamhopkins) [reply packages/edition-slices/src/configured-tiles/tile-col-with-image-bottom/index.tsx:63]: done

File: packages/edition-slices/src/configured-tiles/tile-col-with-image-bottom/index.tsx (hunk 5 above)
Comment (justadamhopkins) [reply packages/edition-slices/src/configured-tiles/tile-col-with-image-bottom/index.tsx:14]: cool you pushed it up on your draft?

File: packages/section/src/utils/transformSlice.ts [hunk 8]
```diff
@@ -36,5 +46,57 @@ export const transformSlice = (isTablet: boolean, sectionTitle: string) => (
       st.name == slice.name &&
       st.sectionTitle.toUpperCase() === sectionTitle.toUpperCase(),
   );
-  return transformation?.transform(slice) ?? slice;
+
+  // no transform object and no default base config so only use slice in old format
+  if (!transformation && !baseConfigs[slice.name]) return slice;
+
+  //merges existing slice tile data with the base config
+  const mergeBaseConfig = Object.keys(slice).reduce((acc, curtileName) => {
```
Comment (aronshamash) [inline packages/section/src/utils/transformSlice.ts:54]: `mergedBaseConfig`

File: packages/section/src/utils/transformSlice.ts [hunk 9]
```diff
@@ -1,27 +1,53 @@
...
+const leadOneAndOneNewsTransform: TransformSlice = {
   sectionTitle: "News",
   name: "LeadOneAndOneSlice",
-  transform: (slice: Slice) => ({
-    ...slice,
-    support: { ...slice.support, config: { showImage: true } },
-  }),
+  overrides: {
+    support: {
+      config: {
```
Comment (aronshamash) [inline packages/section/src/utils/transformSlice.ts:40]: do we need these `config` objects or can we remove ?

File: packages/section/src/utils/transformSlice.ts (hunk 8 above)
Comment (justadamhopkins) [reply packages/section/src/utils/transformSlice.ts]: fixed

File: packages/section/src/utils/transformSlice.ts [hunk 10]
```diff
@@ -36,5 +46,57 @@ export const transformSlice = (isTablet: boolean, sectionTitle: string) => (
...
+
+  // uses base config if someone forgets to set ovverrides in transform
+  if (!transformation.overrides)
+    return {
+      ...slice,
+      ...mergeBaseConfig,
+    };
+
+  //merges existing slice tile data with the transform ovverrides
+  const mergeTileOverrideData = Object.keys(slice).reduce(
```
[the remark this answers is not in the document]
Comment (justadamhopkins) [reply packages/section/src/utils/transformSlice.ts]: mergedTileConfig is good

File: packages/section/src/utils/transformSlice.ts (hunk 1 above)
Comment (justadamhopkins) [reply packages/section/src/utils/transformSlice.ts]: This is still needed I think. Well TS whines when I remove this still

**Pull request closed without merging**